### ISSUE TITLE
Allow assigning None to any type

### ISSUE THREAD

**Issue by Rizhiy** [microsoft/pyright#1805]
**Is your feature request related to a problem? Please describe.**
I believe it is useful to assign None to variable when real value is missing/invalid, but currently pyright shows a `reportGeneralTypeIssues` for such cases.

**Describe the solution you'd like**
Don't show error when None is assigned

**Additional context**
Example:
```python
data = range(10)
results = defaultdict(float)
for item in data:
	if item == 0: # bad item
		# next line would show error
		results[item] = None
	else:
		results[item] = 1 / item
```

**Comment by erictraut** [comment]
The current behavior is correct. It would be a major type violation (and a violation of the PEP 484 specification) to allow `None` to be assigned to a `defaultdict` that is defined as accepting only `float` values.

If you want your `defaultdict` to accept `None` values, you should specify that through the use of a type annotation.

```python
results: DefaultDict[int, Optional[float]] = defaultdict(float)
```
